javaagent-tooling/src/main/java/io/opentelemetry/javaagent/tooling/muzzle/matcher/MuzzleGradlePluginUtil.java

```diff
@@ -54,45 +55,32 @@
   public static void assertInstrumentationMuzzled(
       ClassLoader agentClassLoader, ClassLoader userClassLoader, boolean assertPass)
       throws Exception {
-    // muzzle validate all instrumenters
-    int validatedModulesCount = 0;
-    for (InstrumentationModule instrumentationModule :
-        ServiceLoader.load(InstrumentationModule.class, agentClassLoader)) {
-      ReferenceMatcher muzzle =
-          new ReferenceMatcher(
-              instrumentationModule.getMuzzleHelperClassNames(),
-              instrumentationModule.getMuzzleReferences(),
-              instrumentationModule::isHelperClass);
-      List<Mismatch> mismatches = muzzle.getMismatchedReferenceSources(userClassLoader);
-
-      boolean classLoaderMatch =
-          instrumentationModule.classLoaderMatcher().matches(userClassLoader);
-      boolean passed = mismatches.isEmpty() && classLoaderMatch;
-
-      if (passed && !assertPass) {
-        System.err.println(
-            "MUZZLE PASSED "
-                + instrumentationModule.getClass().getSimpleName()
-                + " BUT FAILURE WAS EXPECTED");
-        throw new IllegalStateException("Instrumentation unexpectedly passed Muzzle validation");
-      } else if (!passed && assertPass) {
-        System.err.println(
-            "FAILED MUZZLE VALIDATION: "
-                + instrumentationModule.getClass().getName()
-                + " mismatches:");
-
-        if (!classLoaderMatch) {
-          System.err.println("-- classloader mismatch");
-        }
 
-        for (Mismatch mismatch : mismatches) {
-          System.err.println("-- " + mismatch);
-        }
-        throw new IllegalStateException("Instrumentation failed Muzzle validation");
-      }
+    ClassLoader contextClassLoader = Thread.currentThread().getContextClassLoader();
+    Thread.currentThread().setContextClassLoader(agentClassLoader);
```

Comment:
setting context class loader always feels mysterious, can you add a comment here explaining, or pass `agentClassLoader` to `matchesAll` and avoid?

Reply:
The main reason I decided not to pass `agentClassLoader` is the simplicity/clarity of the resulting method signature. I admit that relying on context class loader may be perceived as mysterious, but to me this is a "default" mode of operation. I would even prefer to change https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/8c6adb481aa04862acdcf0a7b72bdc4d4d5060a3/buildSrc/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts#L225 and to set `agentClassloader` already there instead of the bogus one.

Reply:
> I would even prefer to change
> 
> https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/8c6adb481aa04862acdcf0a7b72bdc4d4d5060a3/buildSrc/src/main/kotlin/io.opentelemetry.instrumentation.muzzle-check.gradle.kts#L225
> 
> and to set `agentClassloader` already there instead of the bogus one.

oh, I would like that too, does that not work for some reason? (I can try in a separate PR if you want to merge and move on)

Reply:
I have a couple of more PRs in the pipeline :) Will certainly try that.